File: php/7-reverse-integer.php
<?php
class Solution {

    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        $flag = 1;
        if ($x < 0) {
            $flag = -1;
        }
        $s = (string)$x;
        $right = strlen($s) - 1;
        $left = 0;
        while ($left < $right) {
            [$s[$left], $s[$right]] = [$s[$right], $s[$left]];
            $left++;
            $right--;
        }
        $val = intval($s);
        $maxInt32 = (1<<31) - 1;
        return ($val > $maxInt32)? 0 : $flag * $val;
    }
}

File: go/7-reverse-integer.go
package main

func reverse(x int) int {
	flag := 1
	if x < 0 {
		flag = -1
		x = -x
	}
	sx := []byte(strconv.Itoa(x))
	left := 0
	right := len(sx) - 1
	for left < right {
		sx[left], sx[right] = sx[right], sx[left]
		left++
		right--
	}
	number, _ := strconv.Atoi(string(sx))
	if float64(number) > math.Pow(2, 31)-1 || float64(number) < -math.Pow(2, 31) {
		return 0
	}
	return number * flag
}
